### ISSUE TITLE
GHC.Generics in master seems out of date

### ISSUE THREAD

**Issue by adamschoenemann** [typelead/eta#871]
## Description
I'm trying out Eta and therefore trying to build some code with Eta that relies on generics-sop 0.3.2.0 but it's failing due to some imports from `GHC.Generics` that should definitely be in base 4.11.1.0. Looking at [the code in your master][1] it also seems very different from [what's on hackage][2]. Perhaps I'm misunderstanding something completely or doing something wrong (quite likely), but it seems like the recent port of base 4.11.1.0 perhaps is lacking some code?

## Expected Behavior
generics-sop 0.3.2.0 should build

## Actual Behavior
It doesn't, choking on `Meta, MetaData, MetaCons, PrefixI, InfixI, MetaSel`.

## Possible Fix
Port the actual file?

## Steps to Reproduce

1. Checkout and build from master, or use the docker image (which I did)
2. `etlas get generics-sop && cd generics-sop-0.3.2.0`
3. `etlas build`

## Context
Trying to build some proprietary code

## Your Environment

* Did you install an older version of Eta/Etlas before? No, and I'm using docker
* Current Eta & Etlas version: Version 0.8.6b3, Git Revision 61b0641171650373742cf101321bf2c11e9c209a

Thanks!

[1]:https://github.com/typelead/eta/blob/master/libraries/base/GHC/Generics.hs
[2]:http://hackage.haskell.org/package/base-4.11.1.0/docs/src/GHC.Generics.html

**Comment by Jyothsnasrinivas** [comment]
@adamschoenemann Thanks for filing the issue!

When porting `base-4.11` we avoided porting all the new Generics updates to avoid [compile-time regressions](https://ghc.haskell.org/trac/ghc/ticket/5642). Once we have some time to investigate the cause for the slow compile times we'll port them. I filed issue #872 to track the problem.

To fix your problem right now, we've patched `generics-sop-0.3.2.0` to use the old representation. Run `etlas update` and try again :)
